docs/extending/extending_client_side.md

```diff
@@ -0,0 +1,528 @@
+(extending_client_side)=
+
+# Extending client-side behaviour
+
+Many kinds of common customisations can be done without reaching into JavaScript, but depending on what parts of the client-side interaction you want to leverage or customise, you may need to employ React, Stimulus, or vanilla (plain) JS.
+
+[React](https://reactjs.org/) is used for more complex parts of Wagtail, such as the sidebar, commenting system, and the Draftail rich-text editor.
+For basic JavaScript-driven interaction, Wagtail is migrating towards [Stimulus](https://stimulus.hotwired.dev/).
+
+You don't need to know or use these libraries to add your custom behaviour to elements, and in many cases vanilla (plain) JavaScript will work fine, but Stimulus is the recommended approach for anything that vanilla JavaScript can't do.
+
+You don't need to have Node.js tooling running for your custom Wagtail installation for many customisations built on these libraries, but in some cases, such as building packages, it may make more complex development easier.
+
+```{note}
+it's recommended that you avoid using jQuery as this will be removed in a future version of Wagtail.
```

Comment:
If we have this kind of long-term note I feel like we might as well make it a clear-cut "don’t".

```suggestion
Avoid using jQuery and any undocumented jQuery plugins, as they will be removed in a future version of Wagtail.
```